typescript/cli/src/utils/output.ts

```diff
@@ -0,0 +1,53 @@
+import chalk from 'chalk';
+
+export enum ViolationDiffType {
+  None,
+  Expected,
+  Actual,
+}
+
+type FormatterByDiffType = Record<ViolationDiffType, (text: string) => string>;
+
+const defaultDiffFormatter: FormatterByDiffType = {
+  [ViolationDiffType.Actual]: chalk.red,
+  [ViolationDiffType.Expected]: chalk.green,
+  [ViolationDiffType.None]: (text: string) => text,
+};
+
+/**
+ * Takes a yaml formatted string and highlights differences by looking at `expected` and `actual` properties.
+ */
+export function formatYamlViolationsOutput(
+  yamlString: string,
+  formatters: FormatterByDiffType = defaultDiffFormatter,
+): string {
+  const lines = yamlString.split('\n');
+
+  let curr: ViolationDiffType = ViolationDiffType.None;
+  let lastDiffIndent = 0;
+  const highlightedLines = lines.map((line) => {
+    // Get how many white space/tabs we have before the property name
+    const match = line.match(/^(\s*)/);
+    const currentIndent = match ? match[0].length : 0;
+
+    // if the current indentation is smaller than the previous diff one
+    // we just got out of a diff property and we reset the formatting
+    if (currentIndent < lastDiffIndent) {
+      curr = ViolationDiffType.None;
+    }
+
+    if (line.includes('expected:')) {
```

Comment:
nit: can we make this like `(EXPECTED)` so its invalid yaml and clear that it's not a field in the object??